blockstore/apps/bundles/storage.py

```diff
@@ -0,0 +1,165 @@
+"""
+Django Storage backends for bundles app.
+
+If you don't care about the specifics, just import and use `default_asset_storage`,
+which is a `Storage` instance just like `default_storage`.
+"""
+from django.conf import settings
+from django.core.files.storage import Storage, default_storage, get_storage_class
+
+
+# You probably just want to import this.
+# (Forward-declaring here for visibility)
+default_asset_storage: Storage
+
+
+class LongLivedSignedUrlStorage(Storage):  # pylint: disable=abstract-method
+    """
+    Storage backend for generating longer-lived signed URLs.
+
+    It only implements the `url` operation; all other operations (delete,
+    exists, listdir, etc.) will raise NotImplementedErrors.
+    This is why we disable pylint's abstract method check for this class.
+
+    Context:
+
+        When powering Blockstore with the S3SBoto3torageBackend, asset URLs are
+        exposed to clients (and users) by generating signed S3 URLs.
+        The ideal time-to-live of these URLs is ~24-48 hours;
+        that is, long enough such that users may interact with content without
+        losing access to assets in the middle of their session,
+        but short enough as to avoid granting perpetual unauthenticated access
+        to copyrighted content.
+
+        However, servers deployed on AWS are often given *temporary*
+        credentials instead of permanent ones. If those credentials are used
+        to sign S3 URLs, then the S3 URLs will automatically expire when
+        then the server's temporary credentials  expire (even if the URL was just
+        signed!). Thus, temporary credentials, although preferable for security
+        reasons, cannot be depended upon to sign our "long-lasting" S3 URLs.
+
+        The compromise here is to allow Blockstore to have two separate
+        sets of credentials:
+        1. temporary credentials that allow full read/write/delete access to S3
+           (along with any other necessary AWS permissions), and
+        2. permanent credentials that *only* grant read access to S3.
```

Comment:
I think I missed something in all the discussions so far... but why are we going to so much effort to distinguish between temporary credentials and "permanent credentials" ? Why can't we just configure Blockstore to use an `AWS_ACCESS_KEY_ID`/`AWS_SECRET_ACCESS_KEY` for all of its S3 operations, much the same way as we configure it with a "permanent" username+password for MySQL for all RDS operations ?

Using a machine role for S3 auth is what I would consider an unusual approach, and in this case is the source of all these problems, so are the "security reasons" really so compelling as to support introducing this much special-purpose code, instead of just changing the auth config? After all, secret keys can still be rotated any time if needed.

Reply:
@fredsmith , can you speak to this?

Reply:
according to the AWS IAM best practices, temporary credentials (issued to machine roles, via two factor enabled user auth, etc) are preferred over permanent credentials.  We try to follow these best practices wherever possible in our deployments  https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html

> Use temporary security credentials (IAM roles) instead of long-term access keys

Reply:
I see. Can we include that doc + quote in the docstring here?

Reply:
Thanks Fred. And will do.

Reply:
Done.